nomad/vault.go

```diff
@@ -1445,7 +1488,9 @@ func (v *vaultClient) EmitStats(period time.Duration, stopCh <-chan struct{}) {
 // extendExpiration sets the current auth token expiration record to ttLSeconds seconds from now
 func (v *vaultClient) extendExpiration(ttlSeconds int) {
 	v.currentExpirationLock.Lock()
-	v.currentExpiration = time.Now().Add(time.Duration(ttlSeconds) * time.Second)
+	now := time.Now()
+	v.currentExpiration = now.Add(time.Duration(ttlSeconds) * time.Second)
+	v.lastRenewalTime = now
```

Comment:
extendExpiration is [called in renew too.](https://github.com/hashicorp/nomad/blob/f53dba45d51bd6750197e2394b5eaece11e8e067/nomad/vault.go#L686)

Reply:
Yup, but my point is that it should _only_ be called in `renew`, otherwise it will reset the metric value. You would also want to reset it whenever `renew` is called, so this action is tied to the renew process, not necessarily to the expiration extension.